### ISSUE TITLE
FreeBSD DataFolder path

### ISSUE THREAD

**Issue by jacobblock** [Jackett/Jackett#2208]
Hi all,

I recently decided to upgrade Jackett and ran into an apparent regression with the DataFolder CLI option. I saw another issue and a recent change to the help docs suggesting to use XDG_CONFIG_HOME environment variable. This is fine, just a bit awkward/unconventional, but I see in Jackett/src/Jackett.Common/Models/Config/RuntimeSettings.cs that "Jackett" is appended to whatever path is provided for Unix. Users should be able to set an explicit path and that's no longer possible as far as I can see.

For reference, my local /usr/local/etc/rc.d/jackett script:
```
#!/bin/sh
#
# Author: Jacob Block <jacob.block@gmail.com>
#
# $FreeBSD$
#

# PROVIDE: jackett
# REQUIRE: LOGIN
# KEYWORD: shutdown

# Add the following lines to /etc/rc.conf to enable jackett
# jackett_enable="YES"

. /etc/rc.subr

name="jackett"
rcvar=jackett_enable

load_rc_config $name

: ${jackett_enable="NO"}
: ${jackett_user:="jackett"}
: ${jackett_group:="jackett"}
: ${jackett_data_dir:="/usr/local/jackett"}
: ${jackett_port:="9117"}

pidfile="${jackett_data_dir}/jackettconsole.pid"
procname="/usr/local/bin/mono"
command="/usr/sbin/daemon"
command_args="-f ${procname} /usr/local/share/jackett/JackettConsole.exe -p ${jackett_port}"
start_precmd=jackett_precmd

jackett_precmd()
{
        if [ ! -d ${jackett_data_dir} ]; then
                install -d -o ${jackett_user} -g ${jackett_group} ${jackett_data_dir}
        fi
}

run_rc_command "$1"
```

And my local environment modifications, /etc/rc.conf:
```
jackett_enable="YES"
jackett_user="media"
jackett_group="media"
jackett_data_dir="/mnt/apps/jackett/config"
jackett_env="XDG_CONFIG_HOME=${jackett_data_dir}"
```

Jackett version:  0.8.450.0
Mono version (if not using Windows):  4.8.1

Thanks!

**Comment by kaso17** [comment]
The problem is that just changing the data dir is not enough. jackett is using the DataProtection classes which store their keys in $XDG_CONFIG_HOME/.mono/keypairs and there's no way to change this from inside the program. See https://github.com/Jackett/Jackett/issues/1469#issuecomment-325645315
If the extra Jackett directory is a problem for you you can combine the XDG_CONFIG_HOME env var + the datadir CLI option.

**Comment by jacobblock** [comment]
Yes I understand the first part, but combining the XDG_CONFIG_HOME and -d CLI option doesn't seem to be working:

```
env XDG_CONFIG_HOME=/mnt/apps/jackett/config /usr/local/bin/mono /usr/local/share/jackett/JackettConsole.exe -p 9117 -d /mnt/apps/jackett/config -t
```

[log.txt](https://github.com/Jackett/Jackett/files/1517010/log.txt)

I see the early return in DataFolder get method:

```
if (!string.IsNullOrWhiteSpace(this.CustomDataFolder))
{
   return this.CustomDataFolder;
}
```
So I am not sure why I am still seeing the Jackett folder being created and used.

**Comment by kaso17** [comment]
ok, the option got broken during a recent refactoring.
Fixed via 16f255745f735d6ae5ad3629d2d3b2b939902420

**Comment by jacobblock** [comment]
Excellent, thanks!
